sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java

```diff
@@ -200,10 +205,17 @@ public void recognizeEntitiesForEmptyText() {
     @Test
     public void recognizeEntitiesForFaultyText() {
         StepVerifier.create(client.recognizeEntities("!@#%%"))
-            .expectNextCount(0)
+            .assertNext(result -> assertFalse(result.getWarnings().iterator().hasNext()))
```

Comment:
Would it help to assert the warning and confirm result has the expected result + warnings?